bot/tool_executors.py

```
@@ -0,0 +1,1900 @@
+        return "Error: path is required."
+
+    target_path = Path(path_arg)
+    if not target_path.is_absolute():
+        target_path = OUTPUT_DIR / target_path
+
+    # Validate against trusted directories
+    _ensure_trusted_dirs()
+    resolved = target_path.resolve() if target_path.exists() else target_path.parent.resolve() / target_path.name
+    if not any(
+        str(resolved).startswith(str(td.resolve()))
+        for td in TRUSTED_DATA_DIRS
```

Ruff (0.15.12)

[error] Undefined name `TRUSTED_DATA_DIRS`

(F821)